Upgrade packages

peppy opened the pull request:
Mainly for new realm version, but some others along the way.

Not 100% on the SignalR change with exceptions (have not tested yet). I'll test later today if no one beats me.

peppy committed:
Update all packages

peppy committed:
Remove obsoleted serialisation path from signalr exceptions

peppy committed:
Update mismatching translation xmldocs

smoogipoo commented:
Do we know whether the underlying issue causing https://github.com/ppy/osu/issues/27577 has been fixed by now? The realm-core issue is still open.

peppy commented:
> Do we know whether the underlying issue causing #27577 has been fixed by now? The realm-core issue is still open.

Good call, we'll need to test for this before merging.

Joehuu commented:
On my Pixel 6a (Android 14), the game gets stuck on the splash screen, which seems to be the same issue as https://github.com/ppy/osu/discussions/27738 (the last Realm bump `2024.312.0`). Tried `2024.312.0` and it had the same behavior, so nothing changed since.

The game doesn't crash and logcat doesn't seem to output anything helpful.

Also tried on Android 11 and 8 devices + discussion above had an Xperia 5 II (Android 10-12) and they don't have this problem, so it's maybe an Android 14 thing.

peppy commented:
@Joehuu are you able to pull logs from that?

Joehuu commented:
Here's the log when I opened the app and waiting a minute:

[logcat.txt](https://github.com/ppy/osu/files/15053235/logcat.txt)

peppy commented:
Very concerning. Not much of anything useful in the logs as you say.

bdach committed:
Merge branch 'master' into upgrade-packages

bdach committed:
Remove old signature

bdach committed:
Remove redundant constructor

bdach commented:
Pushed merge conflict resolution & code quality fixes here.

> Unsure if the above is related, but on my Pixel 6a (Android 14), the game gets stuck on the splash screen, which seems to be the same issue as #27738 (the last Realm bump `2024.312.0`). Tried `2024.312.0` and it had the same behavior, so nothing changed since.
> (...)
> Also tried on Android 11 and 8 devices + discussion above had an Xperia 5 II (Android 10-12) and they don't have this problem, so it's maybe an Android 14 thing.

Cannot reproduce, game launches fine here. And I _should_ be on android 14.

![Screenshot_2024-04-22-09-40-15-30_fc704e6b13c4fb26bf5e411f75da84f2](https://github.com/ppy/osu/assets/20418176/5c3c490d-341c-4e3a-9d6d-151bd009fedf)

Haven't tested at all besides that.

Joehuu commented:
> Good call, we'll need to test for this before merging. Realm team says [maybe](https://github.com/realm/realm-core/issues/7454#issuecomment-1992623149).

Still getting the same error in example 2 of https://github.com/realm/realm-core/issues/7454 using an exFAT external drive.

Joehuu commented:
Tested again on https://github.com/realm/realm-dotnet/releases/tag/12.1.0 and my two issues (Pixel 6a Android 14 and external exFAT on Windows) still happen.

peppy commented:
Yeah, unfortunately there's been no push to fix the issue yet 😢 .

Joehuu commented:
The external exFAT on Windows issue seems to be fixed in `12.3.0`, but Android 14 on my Pixel 6a still has the same issue.

peppy commented:
@Joehuu has the android issue been reported to realm's end yet?

Joehuu commented:
Not that I know of, no. I'm sure it's realm as just changing realm versions triggers the Android 14 on Pixel issue. It started since https://github.com/realm/realm-dotnet/releases/tag/11.6.1.

Just to see if the external exFAT on Windows issue is related, it is not. That started in https://github.com/realm/realm-dotnet/releases/tag/11.7.0 but as I said, it's fixed in the latest release.

frenzibyte commented:
We...probably should report it to Realm then?

Joehuu commented:
Found the issue, maybe ^.

bdach commented:
Heads up that I might have a possible route forward; see https://github.com/realm/realm-core/issues/7950#issuecomment-2404450706. If realm ends up being fully dead (core *and* sdks) and we end up forking, maybe adopting that cmake patch is going to make things sorta-kinda work for us at least...?

@Joehuu would you be able to test this on device to confirm? You can grab nuget packages with the fix from the artifact listing at https://github.com/bdach/realm-dotnet/actions/runs/11255982167.

Joehuu commented:
Works fine. Did some recommended beatmap downloading and playing/completing a beatmap with no problems.

peppy committed:
Merge branch 'master' into upgrade-packages

peppy committed:
Update realm xmldoc references

peppy committed:
Upgrade more packages again

This also downgrades nunit to be aligned across all projects. Getting it
up-to-date is a bit high effort.

peppy committed:
Appease message pack new inspections

peppy commented:
Realm have fixed the android issue so I've brought this up-to-date for consideration.

Joehuu reviewed (approved):
Everything still seems to be fine after retesting.

smoogipoo commented:
I've done some basic testing for multiplayer, score submission, and beatmap imports. Looks to work. Not willing to spend more time testing this, so let's just go with it and hope nothing breaks.
